Update info about Accumulo support

mosabua opened the pull request.

dain reviewed (approved):
Let's merge this, but we should figure out how to represent this on the website... maybe we need the Attic :)

mosabua commented:
> Let's merge this, but we should figure out how to represent this on the website... maybe we need the Attic :)


This will update the website .. we could move it down from https://trino.io/ecosystem/data-source#official-data-sources to https://trino.io/ecosystem/data-source#other-data-sources .. that would make it more explicit.

Also the docs are deleted just like the code .. hence the link update. I can show you the difference tomorrow @dain before I merge.

mosabua commented:
I thought about this some more based on your input @dain .. given the facts that .. we removed the code, removed the docs, closed all the issues, and are no longer doing anything, I think it is probably more correct to move the data source down to the "Other data sources" .. users can choose to use it by staying with an old Trino version .. but we really wont be doing anything for them.. so its no longer really an "Oofficial data source" .. with the changes I just pushed it looks like that


<img width="1254" alt="image" src="https://github.com/user-attachments/assets/338e072b-4f41-4ca3-8a05-2255ba76d36d" />

mosabua commented on _data/tools.yml:
```diff
@@ -70,20 +70,21 @@
   anchor: apache-accumulo
   category: data-source
-  owner: trinodb
+  owner: other
```
This does the move to the lower section

mosabua commented:
Edited, demoed, and discussed with @dain some more .. approval stands for the current version so I am merging.
